File: Python/Problems/Problems/problem3324.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem3324(Problem):
    def test(self) -> bool:
        target = "abc"

        expected = ["a", "aa", "ab", "aba", "abb", "abc"]

        result = self.stringSequence(target)

        print1DArray(result)

        return expected == result

    def stringSequence(self, target: str) -> List[str]:
        result = []
        curr = ""
        for c in target:
            for i in range(ord('a'), ord(c) + 1):
                result.append(curr + chr(i))

            curr += c

        return result

File: C++/Problems/Problems/problem_3324.cpp
#include "problem_3324.h"

bool problem_3324::test() {
    std::string target = "abc";

    std::vector<std::string> expected { "a", "aa", "ab", "aba", "abb", "abc" };

    auto result = stringSequence(target);

    print1DVector(result);

    return result == expected;
}

std::vector<std::string> problem_3324::stringSequence(std::string target) {
    std::vector<std::string> result;
    std::string curr = "";
    for (auto c : target)
    {
        for (int i = 'a'; i <= c; ++i)
        {
            result.push_back(curr + (char)i);
        }

        curr += c;
    }

    return result;
}
